test_util/test_aws_cf.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+"""Deploys DCOS AWS CF template and then runs integration_test.py
+
+The following environment variables control test procedure:
+
+DCOS_TEMPLATE_URL: string
+    The template to be used for deployment testing
+
+CI_FLAGS: string (default=None)
+    If provided, this string will be passed directly to py.test as in:
+    py.test -vv CI_FLAGS integration_test.py
+
+TEST_ADD_ENV_*: string (default=None)
+    Any number of environment variables can be passed to integration_test.py if
+    prefixed with 'TEST_ADD_ENV_'. The prefix will be removed before passing
+"""
+import logging
+import os
+import random
+import stat
+import string
+import sys
+from contextlib import closing
+
+import test_util.provide_aws
+import test_util.test_runner
+from ssh.ssh_tunnel import SSHTunnel
+
+LOGGING_FORMAT = '[%(asctime)s|%(name)s|%(levelname)s]: %(message)s'
+logging.basicConfig(format=LOGGING_FORMAT, level=logging.DEBUG)
+log = logging.getLogger(__name__)
+
+DEFAULT_AWS_REGION = 'us-west-2'
+
+
+def check_environment():
+    """Test uses environment variables to play nicely with TeamCity config templates
+    Grab all the environment variables here to avoid setting params all over
+
+    Returns:
+        object: generic object used for cleanly passing options through the test
+
+    Raises:
+        AssertionError: if any environment variables or resources are missing
+            or do not conform
+    """
+    options = type('Options', (object,), {})()
+
+    options.variant = os.getenv('DCOS_VARIANT', 'default')
+
+    options.template_url = os.getenv('DCOS_TEMPLATE_URL')
+    assert options.template_url, 'A CF template must be specified!'
+
+    options.ci_flags = os.getenv('CI_FLAGS', '')
+    options.aws_access_key_id = os.getenv('AWS_ACCESS_KEY_ID')
+    options.aws_secret_access_key = os.getenv('AWS_SECRET_ACCESS_KEY')
+    assert options.aws_access_key_id, 'AWS Credentials Required!'
+    assert options.aws_secret_access_key, 'AWS Credentials Required!'
+
+    add_env = {}
+    prefix = 'TEST_ADD_ENV_'
+    for k, v in os.environ.items():
+        if k.startswith(prefix):
+            add_env[k.replace(prefix, '')] = v
+    options.add_env = add_env
+    options.pytest_dir = os.getenv('DCOS_PYTEST_DIR', '/opt/mesosphere/active/dcos-integration-test')
+    options.pytest_cmd = os.getenv('DCOS_PYTEST_CMD', 'py.test -vv '+options.ci_flags)
+    return options
+
+
+def main():
+    options = check_environment()
+
+    random_identifier = ''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(10))
+    unique_cluster_id = "CF-integration-test-{}".format(random_identifier)
+    log.info("Spinning up AWS CloudFormation with ID: {}".format(unique_cluster_id))
+    cf = test_util.provide_aws.SimpleDcosStack(
```

Comment:
When we get this all done, I think this would be something like:
```python
bw = BotoWrapper(aws_region=options.aws_region, aws_access_key_id=options.aws_access_key_id, aws_secret_access_key=options.aws_secret_access_key)
dcos = test_util.aws.DCOSCfSimple.create(name=unique_cluster_id, template_url=options.template_url, private_agents=options.private_agents, public_agents=options.public_agents, admin_location='0.0.0.0/0', key_pair_name='default')
dcos.cf_stack.wait_for_creation()
```